apis/v1/gateway_types.go

```diff
@@ -526,20 +526,30 @@ const (
 
 // GatewayBackendTLS describes backend TLS configuration for gateway.
 type GatewayBackendTLS struct {
-	// ClientCertificateRef is a reference to an object that contains a Client
-	// Certificate and the associated private key.
-	//
-	// References to a resource in different namespace are invalid UNLESS there
-	// is a ReferenceGrant in the target namespace that allows the certificate
-	// to be attached. If a ReferenceGrant does not allow this reference, the
-	// "ResolvedRefs" condition MUST be set to False for this listener with the
-	// "RefNotPermitted" reason.
-	//
-	// ClientCertificateRef can reference to standard Kubernetes resources, i.e.
-	// Secret, or implementation-specific custom resources.
-	//
-	// Support: Core
-	//
+	// ClientCertificateRef references an object that contains a client certificate
+	// and its associated private key. It can reference standard Kubernetes resources,
+	// i.e., Secret, or implementation-specific custom resources.
+	//
+	// A ClientCertificateRef is considered invalid if:
+	//
+	// * It refers to a resource that cannot be resolved (e.g., the referenced resource
+	//   does not exist) or is misconfigured (e.g., a Secret does not contain the keys
+	//   named `tls.crt` and `tls.key`). In this case, the `ResolvedRefs` condition
+	//   on the Gateway MUST be set to False with the Reason `InvalidClientCertificateRef`
```

Comment:
I'm worried that Cx might confused this error with other TLS based features like ClientCertValidation, can we be more specifica about the certificate?  
InvalidClientCertificateRef -> InvalidGatewayClientCertificateRef

Reply:
For simplicity, I’d prefer to keep the reason as `InvalidClientCertificateRef`. Since it’s already set on the Gateway, adding `Gateway` to the reason feels redundant. But if there are better alternatives, I’m happy to adjust.